uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Page/WorkPattern.php

```diff
@@ -27,8 +29,13 @@ public function browse() {
         $rows[$object->id]['number_of_hours'] = ts('Various');
       }
 
+      $links = $this->links();
+
+      if ($this->canNotDelete($object->id)) {
+         unset($links[CRM_Core_Action::DELETE]);
+      }
```

Comment:
This should be done with a mask and not by removing the item from the array. Take a look at the `calculateLinksMask` method in this class